Sources/MuxUploadSDK/InternalUtilities/ChunkedFile.swift

```diff
@@ -85,7 +85,7 @@ class ChunkedFile {
         let data = try fileHandle.read(upToCount: chunkSize)
         guard let data = data else {
             // No more data to read, we reached EOF. The caller shouldn't call us like this, but let's be safe
-            return FileChunk(startByte: 0, endByte: 0, totalFileSize: 0, chunkData: Data(capacity: 0))
+            return FileChunk(startByte: fileSize, endByte: fileSize, totalFileSize: fileSize, chunkData: Data(capacity: 0))
```

Comment:
What do you think of passing the `fileSize` as an argument to `doReadNextChunk`?

Reply:
I don't see what that would gain us? The `ChunkedFile` knows its own size because its job is to do all the chunk math. `ChunkedFile` is the source of truth for how big the file is, where the read position is, and what data is there. I would rather keep it this way

As a side note, my comment here is no longer correct. The assumption that the caller would never call at EOF is the source of the bug, so I have corrected the comment

Reply:
Yeah I think it makes sense for `ChunkedFile` to be the source of truth for how big the file is, where the read position is, and its data content.

The impetus behind that suggestion was that we'd one day might find it useful to move the actual chunk math computation out of `ChunkedFile` (so we can vary it based on various heuristics), but it's IMO not something that's all that useful for now.

Thanks for fixing the comment!